File: solutions/javascript/matrix/1/matrix.js
export class Matrix {
  #data = [];

  constructor(text) {
    this.#data = text
      .split("\n")
      .map((row) => row.split(" ").map((c) => Number(c)));
  }

  get rows() {
    return this.#data;
  }

  get columns() {
    return this.#data[0].map((_, idx) => this.#data.map((row) => row[idx]));
  }
}

File: solutions/java/matrix/1/src/main/java/Matrix.java
import java.util.*;
import java.util.stream.*;

class Matrix {
    private int[][] rowsMajor;

    Matrix(String matrixAsString) {
        this.rowsMajor = matrixAsString
            .lines()
            .map(row -> Arrays.stream(row.split(" ")).mapToInt(Integer::parseInt).toArray())
            .toArray(int[][]::new);       
    }

    int[] getRow(int rowNumber) {
        return this.rowsMajor[rowNumber - 1];
    }

    int[] getColumn(int columnNumber) {
        return Arrays.stream(this.rowsMajor).mapToInt(arr -> arr[columnNumber - 1]).toArray(); 
    }
}
